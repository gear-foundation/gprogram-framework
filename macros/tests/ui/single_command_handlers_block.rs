use gprogram_framework_macros::{command_handlers, gprogram};

#[gprogram]
mod gprogram {
    use super::*;

    command_handlers!(
        fn do_this(value: u32) -> Result<(), ()> {
            Ok(())
        }
    );

    command_handlers!(
        fn do_that(value: u32) -> Result<(), ()> {
            Ok(())
        }
    );
}

fn main() {}
