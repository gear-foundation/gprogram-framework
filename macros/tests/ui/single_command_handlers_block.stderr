error: There must be a single `command_handlers` block in the `gprogram` module
  --> tests/ui/single_command_handlers_block.rs:13:5
   |
13 |     command_handlers!(
   |     ^^^^^^^^^^^^^^^^

warning: unused import: `command_handlers`
 --> tests/ui/single_command_handlers_block.rs:1:33
  |
1 | use gprogram_framework_macros::{command_handlers, gprogram};
  |                                 ^^^^^^^^^^^^^^^^
  |
  = note: `#[warn(unused_imports)]` on by default
